src/code/index.js

```diff
@@ -112,6 +123,49 @@ function previewLink(button) {
     console.log("Placeholder value:", inputElement.placeholder);
     console.log("Input value:", inputValue);
     // Rest of the code...
+    if (isValidURL(inputValue)) {
+        console.log("Valid URL!")
+        button.disabled = false;
+        
+        const matchingKey = Object.keys(socialLinks).find(key => {
+            const inputValueNoUsername = socialLinks[key].replace('<username>', '');
+            return inputValue.startsWith(inputValueNoUsername) && inputValue.length > inputValueNoUsername.length;
+            // ^ This ensures the link has text in the <username> placeholder before returning
+        });
+        if(matchingKey) {
+            console.log("Key found!:", matchingKey);
+        } else {
+            console.log("No key found!");
+        }
+        setIcon(matchingKey, parentElement);
```

Comment:
It's not working, I have tried manually too, not matching with the name of the social handle.
Check screenshots attached.
![Screenshot from 2024-05-17 21-54-33](https://github.com/Param302/SocialRepo/assets/76559816/6ec20a0e-9678-46c6-9d10-b349231fc07c)

Its showing the default icon only.
When `https://linkedin` minimum this keyword should be entered, it should show linkedin icon.

I have bypassed the ValidURL if statement with this:
```js

    if (inputValue.includes("https://linkedin.com")) {
        console.log("Valid URL!")
        button.disabled = false;
        
        const matchingKey = Object.keys(socialLinks).find(key => {
            const inputValueNoUsername = socialLinks[key].replace('<username>', '');
            return inputValue.startsWith(inputValueNoUsername) && inputValue.length > inputValueNoUsername.length;
            // ^ This ensures the link has text in the <username> placeholder before returning
        });
		...
```

Reply:
When you commented above "You can bypass this as of now." you mean don't worry about it anymore, correct?
Because I noticed you added link validation so I deleted mine. Just making sure that I understood correctly.

Ahh I see, I didn't notice that some of the key values started with "www," and others didn't.

```
    let matchingKey;
    const matchingLink = Object.entries(socialLinks).find(([key, value]) => {
        const url = new URL(value.replace("<username>", ""));
        const domainKeyWords = url.hostname.replace("www.", "").replace(".com", "");
        const isMatch = inputValue.includes(domainKeyWords);
```

So instead I'll get the value of the matching socialLink key, take it's domain name and remove `www.` and `.com`,
so that as soon as `https://linkedin` is entered it will retrieve the key and show the icon.
But right now it will still retrieve the key and show the icon if you were to just enter `linkedin` because 
I removed isValidURL.

Let me know if anything needs work, thanks.